config/configauth/authenticator.go

```diff
@@ -17,27 +17,28 @@ package configauth
 import (
 	"context"
 	"errors"
-	"io"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
+
+	"go.opentelemetry.io/collector/component"
 )
 
 var (
-	errNoOIDCProvided   = errors.New("no OIDC information provided")
 	errMetadataNotFound = errors.New("no request metadata found")
-	defaultAttribute    = "authorization"
 )
 
-// Authenticator will authenticate the incoming request/RPC
+// Authenticator is an Extension that can be used as an authenticator for the configauth.Authentication option.
+// Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
+// names from the Authentication configuration. Each Authenticator is free to define its own behavior and configuration options,
+// but note that the expectations that come as part of Extensions exist here as well. For instance, multiple instances of the same
+// authenticator should be possible to exist under different names.
 type Authenticator interface {
-	io.Closer
-
-	// Authenticate checks whether the given context contains valid auth data. Successfully authenticated calls will always return a nil error and a context with the auth data.
-	Authenticate(context.Context, map[string][]string) (context.Context, error)
+	component.Extension
 
-	// Start will
-	Start(context.Context) error
+	// Authenticate checks whether the given context contains valid auth data. Successfully authenticated calls will always return a nil error.
+	// When the authentication fails, an error must be returned and the caller must not retry.
+	Authenticate(context.Context, map[string][]string) error
```

Comment:
What is the purpose of the `Authenticate()` function? As far as I can see it is not called from anywhere except from tests and from the component itself.

Aren't UnaryInterceptor and StreamInterceptor all we need for the functionality that we have?

I assume for HTTP traffic we will also have HTTPInterceptor which does the authentication based on HTTP headers. Is this what `Authenticate()` func is supposed to do?

If that's the case I suggest we explain that `Authenticate()` and the interceptor functions are supposed to do the same thing: "authenticate", and that the `Authenticate()` is the general version that is called when we have an incoming request with a context (including HTTP incoming request and the headers can be extracted from the context), while UnaryInterceptor and StreamInterceptor are special functions for the gRPC incoming connections.

Reply:
I think the description is kind of misleading
> Authenticate checks whether the given context contains valid auth data. Successfully authenticated calls will always return a nil error.

I tought we would check the auth credentials in the headers, if not and only using the context what is the purpose of the map[string][]string?

@tigrannajaryan I think the purpose of Authenticate is to abstract the authentication from the protocol, hence the authenticator will provide the logic to authenticate and the specific interceptors will call this.

Reply:
Right, Authenticate is currently being called by the interceptors, but might potentially be called directly by the receivers, especially if they aren't HTTP/gRPC-based.